### PR TITLE
Full code for esql article

### PR DESCRIPTION
**Pull request by l-trotta**

### CODE REVIEW

File: examples/esql-article/src/main/java/esql.article/EsqlArticle.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            .readValues(new FileReader(csvPath));
+
+        BulkIngester ingester = BulkIngester.of(bi -> bi
+            .client(client)
+            .maxConcurrentRequests(20)
+            .maxOperations(5000));
+
+        boolean hasNext = true;
+
+        while (hasNext) {
```
Comment (swallez) [inline examples/esql-article/src/main/java/esql.article/EsqlArticle.java:124]: Nit: use the more common `while(it.hasNext())` construct?

**Review by swallez — COMMENTED** [review]
Nice example!

Considering that it's a single class, it would probably make sense to move it to `java-client/src/main/test/co/elastic/clients/documentation/api_conventions` so that we can reuse it in ES|QL-related docs.

Or at least make it a child of the root project, so that we don't have a duplicate Gradle installation and so that it can be compiled as part of the regular build process.

File: examples/esql-article/src/main/java/esql.article/EsqlArticle.java (hunk 1 above)
Comment (l-trotta) [reply examples/esql-article/src/main/java/esql.article/EsqlArticle.java:124]: it.hasNext() throws an exception if lines are malformed, I wanted to keep everything into the loop so malformed lines can be ignored

**Review by swallez — APPROVED** [review]
LGTM

**Pull request merged**